### PR TITLE
Add e2e test

### PR DESCRIPTION
**Pull request by outscale-vbr**
Add e2e test:
* Use ginkgo to test (generate clusterctl config, launch clusterctl command using go lib)
* Create contract (https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html)
* Add e2e test in github action
* Change testenv to unit-func-e2e-test
* Only test cluster infrastructure to be provisionned

### CODE REVIEW

**Comment by outscale-vbr** [comment]
Github action: https://github.com/outscale-vbr/cluster-api-provider-outscale/actions/runs/2629221279

File: .github/workflows/unit-func-e2e-test.yaml [hunk 1]
```diff
@@ -200,3 +200,131 @@ jobs:
...
+    - name: Wait to access registry
+      uses: nick-invision/retry@v2
+      with:
+        timeout_seconds: 30
+        max_attempts: 10
+        retry_wait_seconds: 1
+        command: curl http://127.0.0.1:4242/v2
+    - name: Build and Push Docker image 
+      run: |
+        IMG=127.0.0.1:4242/cluster-api-outscale-controller:$IMAGE_TAG make docker-build
```
Comment (outscale-mdr) [inline .github/workflows/unit-func-e2e-test.yaml:284]: This could be put in the `env` section

File: test/e2e/capo.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		Flavor:                   input.ConfigCluster.Flavor,
+		Namespace:                input.ConfigCluster.Namespace,
+		ClusterName:              input.ConfigCluster.ClusterName,
+		KubernetesVersion:        input.ConfigCluster.KubernetesVersion,
+		ControlPlaneMachineCount: input.ConfigCluster.ControlPlaneMachineCount,
+		WorkerMachineCount:       input.ConfigCluster.WorkerMachineCount,
+		InfrastructureProvider:   input.ConfigCluster.InfrastructureProvider,
+		LogFolder:                input.ConfigCluster.LogFolder,
+	})
+	Expect(workloadClusterTemplate).ToNot(BeNil(), "Faiiled to get the cluster template")
```
Comment (outscale-mdr) [inline test/e2e/capo.go:64]: typo: "failed"

File: infrastructure-components.yaml [hunk 3]
```diff
@@ -0,0 +1,833 @@
...
+              key: access_key
+              name: cluster-api-provider-outscale
+              optional: true
+        - name: OSC_SECRET_KEY
+          valueFrom:
+            secretKeyRef:
+              key: secret_key
+              name: cluster-api-provider-outscale
+              optional: true
+        image: 042b4721a38342028d65c28be2b30e64-157001637.eu-west-2.lbu.outscale.com:5000/controller:latest
```
Comment (outscale-mdr) [inline infrastructure-components.yaml:805]: should be dynamic, no ?

File: .github/workflows/unit-func-e2e-test.yaml (hunk 1 above)
Comment (outscale-vbr) [reply .github/workflows/unit-func-e2e-test.yaml]: Ok done

File: test/e2e/capo.go (hunk 2 above)
Comment (outscale-vbr) [reply test/e2e/capo.go]: Ok done

File: infrastructure-components.yaml (hunk 3 above)
Comment (outscale-mdr) [reply infrastructure-components.yaml]: Ok but could you but a real URL here ? Like the future URL in the docker registry ?

Comment (outscale-vbr) [reply infrastructure-components.yaml]: Ok done
values in dockerhub

**Review by jerome-jutteau — APPROVED** [review]

**Comment by outscale-vbr** [comment]
Close #12

**Pull request merged**